pulpcore/app/util.py

```
@@ -188,3 +188,24 @@ def verify_signature(filepath, public_key, detached_data=None):
                 raise InvalidSignatureError(
                     f"The file '{filepath}' does not contain a valid signature."
                 )
+
+
+def touch_content_units(content_units):
+    """Touch the referenced content units in the dict format {"pk": "pulp_href"}.
+
+    If some of the referenced content units are not present in the database, an error with
+    missing content units is raised.
+    """
+    content_units_pks = content_units.keys()
+    existing_content_units = models.Content.objects.filter(pk__in=content_units_pks)
+    existing_content_units.touch()
+
+    existing_content_units_pks = existing_content_units.values_list("pk", flat=True)
+    existing_content_units_pks = {str(pk) for pk in existing_content_units_pks}
+
+    missing_pks = content_units_pks - existing_content_units_pks
```

you are taking away some flexibility here. plugin writers might not want to raise error if some units were not found and follow the best effort approach.
before that the validation was extracted to ` verify_content_units`

---

The error was still raised in the `verify_content_units` method (called from the original `modify` method). The plugin writers had to deal with this behaviour in the past.

With this in mind, it would be probably better to expose the method `verify_content_units` instead of `touch_content_units`. 🦆